lib/puppet/util/posix.rb

```diff
@@ -1,6 +1,39 @@
 # Utility methods for interacting with POSIX objects; mostly user and group
 module Puppet::Util::POSIX
 
+  module GrpLibC
+    def self.user_groups(user)
+      result = []
+      ngroups = 16
```

Comment:
I'm not sure about this, the value 16 seems arbitrary.

My approach would have been to first call `getgrouplist()` with ngroups = 0, use the updated `ngroups` value to allocate the right amount of memory for `groups`, then call `getgrouplist()` again with the correct parameters. 

Calling with ngroups 0 should never _work_ the first time, as the function returns at least 1 group for any queried user. So we can be sure that the first call is to get the correct parameters, and the second call will fill our `groups` array.

I'm open to suggestions though.

Reply:
I like your version better. Implemented it as suggested.